### DISCUSSION TITLE
[Chapter 05] 정적 팩터리 메서드의 네이밍 룰?

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#40] [Chapter 05. REST APIs with Spring MVC]
책의 코드 예시를 보면
``` java
public static HotelRoomIdResponse from(Long hotelRoomId) {
        return new HotelRoomIdResponse(hotelRoomId);
    }

 public static HotelRoomResponse of(Long hotelRoomId, String roomNumber, HotelRoomType hotelRoomType, BigDecimal originalPrice) {
        return new HotelRoomResponse(hotelRoomId, roomNumber, hotelRoomType, originalPrice);
    }
```

정적 팩터리 메서드를 통해 객체를 생성하는 코드가 존재합니다.
그런데 from, of 라는 메서드 도대체 의미를 알 수 없는 메서드 명을 사용하고 있습니다.
- 왜 이러한 메서드명을 사용하는지 궁금합니다.
- 이유가 있다면 이러한 메서드 명과 비슷한 메서드 네이밍의 룰? 이 있는지 궁금합니다.

**Comment by corock** [comment]
정적 팩터리 메서드는 그 유명한 이펙티브 자바 책에서 [아이템 1](https://github.com/orgs/Study-2-Effective-Java/discussions/17) 에서도 언급하고 있죠.
저도 메서드명은 보통 동사형을 사용하는 것으로 알고 있는데, 좀 더 찾아보니 특별한 케이스에는 부분적으로 전치사를 활용하는 것 같습니다.
바로 다른 타입으로 전환하는 메서드나 빌더 패턴을 구현한 클래스의 메서드인 경우인데, 같이 참고하면 좋은 [링크](https://tecoble.techcourse.co.kr/post/2020-04-26-Method-Naming/)를 같이 올려놓을게요.
